packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1911,6 +1911,10 @@ export class MergeTree {
 		let segmentGroup: SegmentGroup | undefined;
 
 		const annotateSegment = (segment: ISegment) => {
+			assert(
+				!(reservedMarkerIdKey in props) || props.markerId === segment.properties?.markerId,
```

Comment:
Also verify this segment is a marker. It should be allowed to set markerId on a segment that is not a marker and refers to a marker.